power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset_handler.hpp

```diff
@@ -31,12 +33,15 @@ struct DatasetInfo {
     std::vector<ComponentInfo> component_info;
 };
 
-template <bool data_mutable, bool indptr_mutable>
-    requires(data_mutable || !indptr_mutable)
+template <data_mutable_tag data_mutable_type, indptr_mutable_tag indptr_mutable_type>
```

Comment:
please LMK if you have a better name for `indptr_mutable_type`

Reply:
Please follow the conversion, a type ending with `_t`, a value ending with `_v`, a type as meta-function without any suffix.

Reply:
how should i work around the name conflict between:

* the concept
* the tag type itself
* the template argument
* the `using` declaration later on

proposal 1:

* the concept                       => suffixed with `_tag`
* the tag type itself               => suffixed with `_t`
* the template argument             => proposed: suffixed with `_`
* the `using` declaration later on  => proposed: without suffix

proposal 2:

* the concept                       => suffixed with `_c`
* the tag type itself               => suffixed with `_t`
* the template argument             => proposed: suffixed with `_tag`
* the `using` declaration later on  => proposed: without suffix

Reply:
Prefer option 1.